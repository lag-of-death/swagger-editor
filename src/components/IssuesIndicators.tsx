import { IRuleResult } from "@stoplight/spectral";
import * as React from "react";
import { connect } from "react-redux";
import styled from "styled-components";
import { displayIssues } from "../actions";
import { dotSize } from "./shared";

const issues = `
  border-radius: 360px;
  width: ${dotSize};
  height: 100%;

  &:hover {
    cursor: pointer;
  }
`;

const Errors = styled.button`
  ${issues}
  background: #9F0000;
`;
const Warnings = styled.button`
  ${issues}
  background: #FF8C00;
  margin-left: 6px;
`;

interface IIssuesIndicatorsProps {
  warnings: IRuleResult[];
  errors: IRuleResult[];
  onClick: (type: string, msg: string) => null;
}

const IssuesIndicators = ({warnings, errors, onClick}: IIssuesIndicatorsProps) => {
  return (
    <React.Fragment>
      {errors.length
        ? <Errors onClick={() => onClick("errors", JSON.stringify(errors))}/>
        : null
      }
      {warnings.length
        ? <Warnings onClick={() => onClick("warnings", JSON.stringify(warnings))}/>
        : null
      }
    </React.Fragment>
  );
};

export default connect(() => ({}), (dispatch) => {
  return {
    onClick: (issueType: string, msg: string) => {
      dispatch(displayIssues(issueType, msg));
    },
  };
})(IssuesIndicators);
