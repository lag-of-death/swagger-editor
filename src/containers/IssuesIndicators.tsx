import * as React from "react";
import { connect } from "react-redux";
import styled from "styled-components";
import { displayIssues } from "../actions";
import { dotSize } from "../components/shared";

const issues = `
  border-radius: 360px;
  width: ${dotSize};
  height: 100%;

  &:hover {
    cursor: pointer;
  }
`;

const Errors = styled.button`
  ${issues}
  background: #9F0000;
`;
const Warnings = styled.button`
  ${issues}
  background: #FF8C00;
  margin-left: 6px;
`;

interface IIssuesIndicatorsProps {
  warnings: Array<{ message: string, path: string[] }>;
  errors: Array<{ message: string, path: string[] }>;
  onClick: (msg: Array<{ message: string, path: string[] }>) => null;
}

export const IssuesIndicators = ({warnings, errors, onClick}: IIssuesIndicatorsProps) => {
  return (
    <React.Fragment>
      {errors.length
        ? <Errors onClick={() => onClick(errors)}/>
        : null
      }
      {warnings.length
        ? <Warnings onClick={() => onClick(warnings)}/>
        : null
      }
    </React.Fragment>
  );
};

export default connect(() => ({}), (dispatch) => {
  return {
    onClick: (msg: Array<{ message: string, path: string[] }>) => {
      dispatch(displayIssues(msg));
    },
  };
})(IssuesIndicators);
