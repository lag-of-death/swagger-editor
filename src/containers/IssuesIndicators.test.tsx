import { shallow } from "enzyme";
import * as React from "react";
import { IssuesIndicators } from "./IssuesIndicators";

describe("IssuesIndicators", () => {
  beforeAll(() => {
    this.onClick = jest.fn();

    this.component = shallow(
      <IssuesIndicators
        warnings={[{message: "warn", path: ["a", "b"]}]}
        errors={[{message: "error", path: ["a", "b", "c"]}]}
        onClick={this.onClick}/>,
    );
  });

  test("rendering", () => {
    expect(this.component).toMatchSnapshot();
  });

  test("event handler", () => {
    this.component.childAt(0).simulate("click");
    this.component.childAt(1).simulate("click");

    expect(this.component).toMatchSnapshot();

    expect(this.onClick).toBeCalledWith([{message: "error", path: ["a", "b", "c"]}]);
    expect(this.onClick).toBeCalledWith([{message: "warn", path: ["a", "b"]}]);
  });
});
