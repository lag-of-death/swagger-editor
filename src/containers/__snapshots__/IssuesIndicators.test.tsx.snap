// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IssuesIndicators event handler 1`] = `
<Fragment>
  <styled.button
    onClick={[Function]}
  />
  <styled.button
    onClick={[Function]}
  />
</Fragment>
`;

exports[`IssuesIndicators rendering 1`] = `
<Fragment>
  <styled.button
    onClick={[Function]}
  />
  <styled.button
    onClick={[Function]}
  />
</Fragment>
`;
